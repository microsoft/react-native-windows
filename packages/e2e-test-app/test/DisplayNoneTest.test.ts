/**
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * @format
 */

import {goToComponentExample, dumpVisualTree} from './framework';

beforeAll(async () => {
  await goToComponentExample('Display:none Style');
});

describe('DisplayNoneTest', () => {
  /* Test case #1: display:none disabled */
  test('DisplayNoneDisabledTest', async () => {
    const dump = await dumpVisualTree('textbox-container');
    expect(dump).toMatchSnapshot();
  });

  /* Test case #2: Enable display:none once, update view*/
  test('DisplayNoneEnabledTest', async () => {
    await toggleDisplayNone();
    const dump = await dumpVisualTree('textbox-container');
    expect(dump).toMatchSnapshot();
  });
});

async function toggleDisplayNone() {
  const showDisplayNoneToggle = await $('~toggle-display:none');
  await showDisplayNoneToggle.click();
}
