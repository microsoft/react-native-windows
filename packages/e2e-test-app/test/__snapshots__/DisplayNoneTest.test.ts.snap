// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayNoneTest DisplayNoneDisabledTest 1`] = `
Object {
  "AutomationId": "textbox-container",
  "Background": null,
  "BorderBrush": "#00000000",
  "BorderThickness": "0,0,0,0",
  "Clip": null,
  "CornerRadius": "0,0,0,0",
  "FlowDirection": "LeftToRight",
  "Height": 40,
  "HorizontalAlignment": "Stretch",
  "Left": 0,
  "Margin": "0,0,0,0",
  "Top": 50,
  "VerticalAlignment": "Stretch",
  "Visibility": "Visible",
  "Width": 718,
  "XamlType": "Microsoft.ReactNative.ViewPanel",
  "children": Array [
    Object {
      "Background": "#B3FFFFFF",
      "BorderBrush": "#FF808080",
      "BorderThickness": "1,1,1,1",
      "Clip": null,
      "CornerRadius": "0,0,0,0",
      "FlowDirection": "LeftToRight",
      "Foreground": "#E4000000",
      "Height": 40,
      "HorizontalAlignment": "Stretch",
      "Left": 0,
      "Margin": "0,0,0,0",
      "Padding": "10,5,6,6",
      "Text": "TextInput...",
      "Top": 0,
      "VerticalAlignment": "Stretch",
      "Visibility": "Visible",
      "Width": 718,
      "XamlType": "Windows.UI.Xaml.Controls.TextBox",
      "children": Array [
        Object {
          "Background": null,
          "BorderBrush": null,
          "BorderThickness": "0,0,0,0",
          "Clip": null,
          "CornerRadius": "0,0,0,0",
          "FlowDirection": "LeftToRight",
          "HorizontalAlignment": "Stretch",
          "Left": 0,
          "Margin": "0,0,0,0",
          "Padding": "0,0,0,0",
          "Top": 0,
          "VerticalAlignment": "Stretch",
          "Visibility": "Visible",
          "XamlType": "Windows.UI.Xaml.Controls.Grid",
          "children": Array [
            Object {
              "Background": "#B3FFFFFF",
              "BorderBrush": "#FF808080",
              "BorderThickness": "1,1,1,1",
              "Clip": null,
              "CornerRadius": "0,0,0,0",
              "FlowDirection": "LeftToRight",
              "HorizontalAlignment": "Stretch",
              "Left": 0,
              "Margin": "0,0,0,0",
              "Name": "BorderElement",
              "Padding": "0,0,0,0",
              "Top": 0,
              "VerticalAlignment": "Stretch",
              "Visibility": "Visible",
              "XamlType": "Windows.UI.Xaml.Controls.Border",
            },
            Object {
              "Background": "#00FFFFFF",
              "BorderBrush": "#00FFFFFF",
              "BorderThickness": "0,0,0,0",
              "Clip": null,
              "CornerRadius": "0,0,0,0",
              "FlowDirection": "LeftToRight",
              "Foreground": "#E4000000",
              "HorizontalAlignment": "Stretch",
              "Left": 0,
              "Margin": "1,1,1,1",
              "Name": "ContentElement",
              "Padding": "10,5,6,6",
              "Top": 0,
              "VerticalAlignment": "Stretch",
              "Visibility": "Visible",
              "XamlType": "Windows.UI.Xaml.Controls.ScrollViewer",
              "children": Array [
                Object {
                  "Background": null,
                  "BorderBrush": "#00FFFFFF",
                  "BorderThickness": "0,0,0,0",
                  "Clip": null,
                  "CornerRadius": "0,0,0,0",
                  "FlowDirection": "LeftToRight",
                  "HorizontalAlignment": "Stretch",
                  "Left": 0,
                  "Margin": "0,0,0,0",
                  "Name": "Root",
                  "Padding": "0,0,0,0",
                  "Top": 0,
                  "VerticalAlignment": "Stretch",
                  "Visibility": "Visible",
                  "XamlType": "Windows.UI.Xaml.Controls.Border",
                  "children": Array [
                    Object {
                      "Background": "#00FFFFFF",
                      "BorderBrush": null,
                      "BorderThickness": "0,0,0,0",
                      "Clip": null,
                      "CornerRadius": "0,0,0,0",
                      "FlowDirection": "LeftToRight",
                      "HorizontalAlignment": "Stretch",
                      "Left": 0,
                      "Margin": "0,0,0,0",
                      "Padding": "0,0,0,0",
                      "Top": 0,
                      "VerticalAlignment": "Stretch",
                      "Visibility": "Visible",
                      "XamlType": "Windows.UI.Xaml.Controls.Grid",
                      "children": Array [
                        Object {
                          "Background": null,
                          "BorderBrush": null,
                          "BorderThickness": "0,0,0,0",
                          "Clip": "Windows.UI.Xaml.Media.RectangleGeometry",
                          "CornerRadius": "0,0,0,0",
                          "FlowDirection": "LeftToRight",
                          "Foreground": "#E4000000",
                          "HorizontalAlignment": "Stretch",
                          "Left": 0,
                          "Margin": "10,5,6,6",
                          "Name": "ScrollContentPresenter",
                          "Padding": "0,0,0,0",
                          "Top": 0,
                          "VerticalAlignment": "Stretch",
                          "Visibility": "Visible",
                          "XamlType": "Windows.UI.Xaml.Controls.ScrollContentPresenter",
                          "children": Array [
                            Object {
                              "Clip": null,
                              "FlowDirection": "LeftToRight",
                              "HorizontalAlignment": "Stretch",
                              "Left": 0,
                              "Margin": "0,0,0,0",
                              "Top": 0,
                              "VerticalAlignment": "Stretch",
                              "Visibility": "Visible",
                              "XamlType": "Windows.UI.Xaml.FrameworkElement",
                            },
                          ],
                        },
                        Object {
                          "Background": null,
                          "BorderBrush": null,
                          "BorderThickness": "0,0,0,0",
                          "Clip": null,
                          "CornerRadius": "0,0,0,0",
                          "FlowDirection": "LeftToRight",
                          "HorizontalAlignment": "Stretch",
                          "Left": 0,
                          "Margin": "0,0,0,0",
                          "Padding": "0,0,0,0",
                          "Top": 0,
                          "VerticalAlignment": "Stretch",
                          "Visibility": "Visible",
                          "XamlType": "Windows.UI.Xaml.Controls.Grid",
                        },
                      ],
                    },
                  ],
                },
              ],
            },
            Object {
              "Background": null,
              "BorderBrush": null,
              "BorderThickness": "0,0,0,0",
              "Clip": null,
              "CornerRadius": "0,0,0,0",
              "FlowDirection": "LeftToRight",
              "Foreground": "#99000000",
              "HorizontalAlignment": "Stretch",
              "Left": 0,
              "Margin": "0,0,0,0",
              "Name": "DescriptionPresenter",
              "Padding": "0,0,0,0",
              "Top": 0,
              "VerticalAlignment": "Stretch",
              "Visibility": "Visible",
              "XamlType": "Windows.UI.Xaml.Controls.ContentPresenter",
            },
          ],
        },
      ],
    },
  ],
}
`;

exports[`DisplayNoneTest DisplayNoneEnabledTest 1`] = `
Object {
  "AutomationId": "textbox-container",
  "Background": null,
  "BorderBrush": "#00000000",
  "BorderThickness": "0,0,0,0",
  "Clip": null,
  "CornerRadius": "0,0,0,0",
  "FlowDirection": "LeftToRight",
  "Height": 0,
  "HorizontalAlignment": "Stretch",
  "Left": 0,
  "Margin": "0,0,0,0",
  "Top": 0,
  "VerticalAlignment": "Stretch",
  "Visibility": "Collapsed",
  "Width": 0,
  "XamlType": "Microsoft.ReactNative.ViewPanel",
  "children": Array [
    Object {
      "Background": "#B3FFFFFF",
      "BorderBrush": "#FF808080",
      "BorderThickness": "1,1,1,1",
      "Clip": null,
      "CornerRadius": "0,0,0,0",
      "FlowDirection": "LeftToRight",
      "Foreground": "#E4000000",
      "Height": 0,
      "HorizontalAlignment": "Stretch",
      "Left": 0,
      "Margin": "0,0,0,0",
      "Padding": "10,5,6,6",
      "Text": "TextInput...",
      "Top": 0,
      "VerticalAlignment": "Stretch",
      "Visibility": "Visible",
      "Width": 0,
      "XamlType": "Windows.UI.Xaml.Controls.TextBox",
      "children": Array [
        Object {
          "Background": null,
          "BorderBrush": null,
          "BorderThickness": "0,0,0,0",
          "Clip": null,
          "CornerRadius": "0,0,0,0",
          "FlowDirection": "LeftToRight",
          "HorizontalAlignment": "Stretch",
          "Left": 0,
          "Margin": "0,0,0,0",
          "Padding": "0,0,0,0",
          "Top": 0,
          "VerticalAlignment": "Stretch",
          "Visibility": "Visible",
          "XamlType": "Windows.UI.Xaml.Controls.Grid",
          "children": Array [
            Object {
              "Background": "#B3FFFFFF",
              "BorderBrush": "#FF808080",
              "BorderThickness": "1,1,1,1",
              "Clip": null,
              "CornerRadius": "0,0,0,0",
              "FlowDirection": "LeftToRight",
              "HorizontalAlignment": "Stretch",
              "Left": 0,
              "Margin": "0,0,0,0",
              "Name": "BorderElement",
              "Padding": "0,0,0,0",
              "Top": 0,
              "VerticalAlignment": "Stretch",
              "Visibility": "Visible",
              "XamlType": "Windows.UI.Xaml.Controls.Border",
            },
            Object {
              "Background": "#00FFFFFF",
              "BorderBrush": "#00FFFFFF",
              "BorderThickness": "0,0,0,0",
              "Clip": null,
              "CornerRadius": "0,0,0,0",
              "FlowDirection": "LeftToRight",
              "Foreground": "#E4000000",
              "HorizontalAlignment": "Stretch",
              "Left": 0,
              "Margin": "1,1,1,1",
              "Name": "ContentElement",
              "Padding": "10,5,6,6",
              "Top": 0,
              "VerticalAlignment": "Stretch",
              "Visibility": "Visible",
              "XamlType": "Windows.UI.Xaml.Controls.ScrollViewer",
              "children": Array [
                Object {
                  "Background": null,
                  "BorderBrush": "#00FFFFFF",
                  "BorderThickness": "0,0,0,0",
                  "Clip": null,
                  "CornerRadius": "0,0,0,0",
                  "FlowDirection": "LeftToRight",
                  "HorizontalAlignment": "Stretch",
                  "Left": 0,
                  "Margin": "0,0,0,0",
                  "Name": "Root",
                  "Padding": "0,0,0,0",
                  "Top": 0,
                  "VerticalAlignment": "Stretch",
                  "Visibility": "Visible",
                  "XamlType": "Windows.UI.Xaml.Controls.Border",
                  "children": Array [
                    Object {
                      "Background": "#00FFFFFF",
                      "BorderBrush": null,
                      "BorderThickness": "0,0,0,0",
                      "Clip": null,
                      "CornerRadius": "0,0,0,0",
                      "FlowDirection": "LeftToRight",
                      "HorizontalAlignment": "Stretch",
                      "Left": 0,
                      "Margin": "0,0,0,0",
                      "Padding": "0,0,0,0",
                      "Top": 0,
                      "VerticalAlignment": "Stretch",
                      "Visibility": "Visible",
                      "XamlType": "Windows.UI.Xaml.Controls.Grid",
                      "children": Array [
                        Object {
                          "Background": null,
                          "BorderBrush": null,
                          "BorderThickness": "0,0,0,0",
                          "Clip": "Windows.UI.Xaml.Media.RectangleGeometry",
                          "CornerRadius": "0,0,0,0",
                          "FlowDirection": "LeftToRight",
                          "Foreground": "#E4000000",
                          "HorizontalAlignment": "Stretch",
                          "Left": 0,
                          "Margin": "10,5,6,6",
                          "Name": "ScrollContentPresenter",
                          "Padding": "0,0,0,0",
                          "Top": 0,
                          "VerticalAlignment": "Stretch",
                          "Visibility": "Visible",
                          "XamlType": "Windows.UI.Xaml.Controls.ScrollContentPresenter",
                          "children": Array [
                            Object {
                              "Clip": null,
                              "FlowDirection": "LeftToRight",
                              "HorizontalAlignment": "Stretch",
                              "Left": 0,
                              "Margin": "0,0,0,0",
                              "Top": 0,
                              "VerticalAlignment": "Stretch",
                              "Visibility": "Visible",
                              "XamlType": "Windows.UI.Xaml.FrameworkElement",
                            },
                          ],
                        },
                        Object {
                          "Background": null,
                          "BorderBrush": null,
                          "BorderThickness": "0,0,0,0",
                          "Clip": null,
                          "CornerRadius": "0,0,0,0",
                          "FlowDirection": "LeftToRight",
                          "HorizontalAlignment": "Stretch",
                          "Left": 0,
                          "Margin": "0,0,0,0",
                          "Padding": "0,0,0,0",
                          "Top": 0,
                          "VerticalAlignment": "Stretch",
                          "Visibility": "Visible",
                          "XamlType": "Windows.UI.Xaml.Controls.Grid",
                        },
                      ],
                    },
                  ],
                },
              ],
            },
            Object {
              "Background": null,
              "BorderBrush": null,
              "BorderThickness": "0,0,0,0",
              "Clip": null,
              "CornerRadius": "0,0,0,0",
              "FlowDirection": "LeftToRight",
              "Foreground": "#99000000",
              "HorizontalAlignment": "Stretch",
              "Left": 0,
              "Margin": "0,0,0,0",
              "Name": "DescriptionPresenter",
              "Padding": "0,0,0,0",
              "Top": 0,
              "VerticalAlignment": "Stretch",
              "Visibility": "Visible",
              "XamlType": "Windows.UI.Xaml.Controls.ContentPresenter",
            },
          ],
        },
      ],
    },
  ],
}
`;
